xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/attachment/XWikiAttachmentAccessWrapper.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiAttachment;
+
+/**
+ * Attachment wrapper for {@link XWikiAttachment}. Provides access to various metadata required to access an attachment.
+ * For instance, its size, name, and an input steam of its content.
+ *
+ * @version $Id$
+ * @since 14.10RC1
+ */
+@Unstable
+public class XWikiAttachmentAccessWrapper implements AttachmentAccessWrapper
```

Comment:
Do you need those implementations to be API? Wondering if it shouldn't stay internal.

Reply:
@tmortagne wdyt. iirc you mentioned reusing this of replication, and in this case I guess it would be useful to already have a wrapper available.

Reply:
> you mentioned reusing this for replication

Hmm, does not ring a bell, actually. Do you remember when that was ?

Right now, I'm tempted to agree with @surli.

Reply:
> Hmm, does not ring a bell, actually. Do you remember when that was ?

Can't find it, I'm probably mixing things up.

Making those internal then!